de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/editor/LocalEditorStatusChangeHandler.java

```diff
@@ -0,0 +1,188 @@
+package de.fu_berlin.inf.dpp.intellij.editor;
+
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.fileEditor.FileEditorManager;
+import com.intellij.openapi.fileEditor.FileEditorManagerEvent;
+import com.intellij.openapi.fileEditor.FileEditorManagerListener;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.util.messages.MessageBusConnection;
+import de.fu_berlin.inf.dpp.activities.SPath;
+import de.fu_berlin.inf.dpp.filesystem.IFile;
+import de.fu_berlin.inf.dpp.intellij.editor.annotations.AnnotationManager;
+import de.fu_berlin.inf.dpp.intellij.filesystem.VirtualFileConverter;
+import de.fu_berlin.inf.dpp.intellij.session.SessionUtils;
+import org.apache.log4j.Logger;
+import org.jetbrains.annotations.NotNull;
+
+/** Dispatches activities for editor changes. */
+class LocalEditorStatusChangeHandler implements DisableableHandler {
+
+  private static final Logger LOG = Logger.getLogger(LocalEditorStatusChangeHandler.class);
+
+  private final LocalEditorHandler localEditorHandler;
+  private final AnnotationManager annotationManager;
+
+  private MessageBusConnection messageBusConnection;
+  private boolean enabled;
+
+  private final FileEditorManagerListener fileEditorManagerListener =
+      new FileEditorManagerListener() {
+        @Override
+        public void fileOpened(@NotNull FileEditorManager source, @NotNull VirtualFile file) {
+          setUpOpenedEditor(file);
+        }
+
+        @Override
+        public void fileClosed(@NotNull FileEditorManager source, @NotNull VirtualFile file) {
+          generateEditorClosedActivity(file);
+        }
+
+        @Override
+        public void selectionChanged(@NotNull FileEditorManagerEvent event) {
+          generateEditorActivatedActivity(event);
+        }
+      };
+
+  private final FileEditorManagerListener.Before beforeFileEditorManagerListener =
+      new FileEditorManagerListener.Before() {
+        @Override
+        public void beforeFileClosed(@NotNull FileEditorManager source, @NotNull VirtualFile file) {
+          cleanUpAnnotations(file);
+        }
+
+        /**
+         * NOP. Only needed to preserve backwards compatibility to Intellij versions older than
+         * 2018.2.6.
+         */
+        // TODO remove once requiring the users to use Intellij 2018.2.6 or newer is acceptable
+        @Override
+        public void beforeFileOpened(@NotNull FileEditorManager source, @NotNull VirtualFile file) {
```

Comment:
As the comment already states, this NOP implementation is needed to keep backwards compatibility. When this PR is merged, I will create an issue tracking such workarounds until they can be removed.